vaadin-date-time-picker-flow/src/main/java/com/vaadin/flow/component/datetimepicker/DateTimePicker.java

```diff
@@ -0,0 +1,233 @@
+        this(initialDateTime);
+        addValueChangeListener(listener);
+    }
+
+    /**
+     * Convenience Constructor to create a date time picker with pre-selected date
+     * and locale setup.
+     *
+     * @param initialDateTime
+     *            the pre-selected date time in the picker
+     * @param locale
+     *            the locale for the date time picker
+     */
+    public DateTimePicker(LocalDateTime initialDateTime, Locale locale) {
+        this(initialDateTime);
+        setLocale(locale);
+    }
+
+    /**
+     * This method should return the value of the field, based on value of the internal fields.
+     *
+     * @return new value of the field.
+     */
+    protected LocalDateTime generateModelValue() {
+        final LocalDate date = datePicker.getValue();
+        final LocalTime time = timePicker.getValue();
+        return date != null && time != null ?
+                LocalDateTime.of(date, time) :
+                null;
+    }
+
+    @Override
+    protected void setPresentationValue(
+            LocalDateTime newPresentationValue) {
+        datePicker.setValue(newPresentationValue != null ?
+                newPresentationValue.toLocalDate() :
+                null);
+        timePicker.setValue(newPresentationValue != null ?
+                newPresentationValue.toLocalTime() :
+                null);
+    }
+
+    @Override
+    protected void setModelValue(LocalDateTime newModelValue, boolean fromClient) {
+        super.setModelValue(newModelValue, fromClient);
+    }
+
+    /**
+     * Regenerates the value by calling {@link #generateModelValue()} and updates the model.
+     * If the value is different than the current one, a {@link ValueChangeEvent} will be generated with {@link ValueChangeEvent#isFromClient()} set to <code>true</code>
+     *
+     * <p>This method is called when the web component generates a <b>changed</b> event, typically in response to a change made by the user in one of the contained fields.
+     *
+     * <p>Custom implementations of this method <b>must</b> call {@link #setModelValue(LocalDateTime, boolean)} with the updated model value.
+     * Subclasses can call this method when the model value needs to be regenerated and updated.
+     */
+    protected void updateValue() {
+        // The second parameter is true since this method is called in response to a change event from the client.
+        setModelValue(generateModelValue(), true);
+    }
+
+    /**
+     * Adds the given Component to the specified slot of this component.
+     *
+     * @param component The component to add.
+     * @param slot The name of the slot to add the component to.
+     * @see <a
+     *      href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot">MDN
+     *      page about slots</a>
+     * @see <a
+     *      href="https://html.spec.whatwg.org/multipage/scripting.html#the-slot-element">Spec
+     *      website about slots</a>
+     */
```

Comment:
Private methods don't need JavaDocs, especially when the implementation is trivial as here and easy to understand just by reading the code.

Even when the implementation is complex and needs comments, I usually prefer normal comment instead of JavaDocs, so I'm not forced to document the obvious parameters etc. which are required in public documentation.

But I guess it's ok to have it here now that it's already done. It could cause confusion because at a first glance it looks like a public method because of having this long documentation.